File: 2024/09/05/4. Median of Two Sorted Arrays/solution.cpp
class Solution {
public:
    double findMedianSortedArrays(vector<int>& nums1, vector<int>& nums2) {
        int n = nums1.size()-1;
        int m = nums2.size()-1;
        vector<int> res;
        for(int i = 0; i<nums1.size();i++){
            res.push_back(nums1[i]);
        }
        for(int i = 0; i<nums2.size();i++){
            res.push_back(nums2[i]);
        }
        sort(res.begin(),res.end());
        if (res.size() % 2 == 0) {
            return (float)(res[(res.size() / 2)] + res[((res.size() / 2) - 1)]) / 2;
        } else {
            return res[res.size() / 2];
        }
    }
};

File: 2024/09/05/4. Median of Two Sorted Arrays/solution.py
class Solution(object):
    def findMedianSortedArrays(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: float
        """
        new_list=[]
        while len(nums1) is not 0 and len(nums2) is not 0:
            if nums1[0]<= nums2[0]:
                new_list.append(nums1.pop(0))
            else:
                new_list.append(nums2.pop(0))
        while len(nums1) is not 0:
            new_list.append(nums1.pop(0))
        while len(nums2) is not 0:
            new_list.append(nums2.pop(0))
        if len(new_list)%2==0:
            return (new_list[len(new_list)/2-1]+new_list[len(new_list)/2])/2.0
        else:
            return new_list[len(new_list)/2]
        
